dbms/src/Core/SortDescription.h

```diff
@@ -20,12 +20,25 @@ struct SortColumnDescription
     int nulls_direction;     /// 1 - NULLs and NaNs are greater, -1 - less.
                              /// To achieve NULLS LAST, set it equal to direction, to achieve NULLS FIRST, set it opposite.
     std::shared_ptr<Collator> collator; /// Collator for locale-specific comparison of strings
-
-    SortColumnDescription(size_t column_number_, int direction_, int nulls_direction_, const std::shared_ptr<Collator> & collator_ = nullptr)
-        : column_number(column_number_), direction(direction_), nulls_direction(nulls_direction_), collator(collator_) {}
-
-    SortColumnDescription(const std::string & column_name_, int direction_, int nulls_direction_, const std::shared_ptr<Collator> & collator_ = nullptr)
-        : column_name(column_name_), column_number(0), direction(direction_), nulls_direction(nulls_direction_), collator(collator_) {}
+    bool with_fill;          /// If true, all missed values in range [FROM, TO] will be filled
+                             /// Range [FROM, TO] respects sorting direction
+    double fill_from;             /// Fill value >= FROM
```

Comment:
`double` is not suitable for large 64bit integers as well as decimal numbers.
You should use Field.